### PR TITLE
feat: button

### PR DESCRIPTION
**Pull request by krishanu2911**
Added button component with the variation of
1. Primary
2. Secondary 
3. link
4. success
5. fail
6. icon
7. floating action button.

### CODE REVIEW

**Comment by krishanu2911** [comment]
https://user-images.githubusercontent.com/75560307/152445675-4ce7581f-ced8-4610-8878-5660d82d3e11.mp4

File: CSS/components/button.css [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    color: var(--fail-color-light);
+}
+.btn-icon{
+    padding: 1rem;
+}
+.fab-div{
+    position: relative;
+}
+.fab{
+    min-width: 4rem;
```
Comment (vivekmehto) [inline CSS/components/button.css:51]: fab should have a position fixed relative to the body.

File: CSS/components/button.css [hunk 2]
```diff
@@ -0,0 +1,57 @@
+.btn{
+    min-width: 5rem;
```
Comment (vivekmehto) [inline CSS/components/button.css]: is there any specific reason for using min-width or min-height?

File: CSS/components/button.css [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    min-height: 1.5rem;
```
Comment (vivekmehto) [inline CSS/components/button.css]: you can use width max-content just a suggestion.

**Review by vivekmehto — COMMENTED** [review]
Suggested some changes.

File: CSS/color.css [hunk 4]
```diff
@@ -13,4 +13,8 @@
     --gray: hsl(217, 19%, 27%);
     --light-gray: hsl(220, 9%, 46%);
     --lightest-gray: hsl(218, 11%, 65%);
+    --success-color: #22c55e;
+    --success-color-light: #bbf7d0;
+    --fail-color: #ef4444;
+    --fail-color-light:#fecaca;
```
Comment (ttahm3d) [inline CSS/color.css]: From the attached screenshot / video, your colors are not clearly differentiable (bg and font color). Try to increase the contrast

File: CSS/components/button.css [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    padding: 0.5rem;
+    background-color: var(--cta);
+    color: var(--button-text-color);
+    border-radius: 0.2rem;
+}
+.btn i{
+    font-size: 1.5rem;
+}
+.btn-secondary{
+    border: 3px solid var(--cta);
```
Comment (ttahm3d) [inline CSS/components/button.css]: Dont define color name as CTA. Use words such as primary to define

**Review by ttahm3d — COMMENTED** [review]
Change colors. They are not accessible enough

File: CSS/components/button.css (hunk 1 above)
Comment (krishanu2911) [reply CSS/components/button.css:51]: I was also trying to use the fixed the btn but I think if we leave users to decide the position accordingly to their way of organising the HTML will be fine bz I was having this problem while testing 
I forget to remove the .fab-div will remove it thank you for pointing out

File: CSS/components/button.css (hunk 3 above)
Comment (krishanu2911) [reply CSS/components/button.css]: Ya refactored the code and used max-content
Got to know about this new property thank you

File: CSS/components/button.css (hunk 5 above)
Comment (krishanu2911) [reply CSS/components/button.css]: changed the color name

File: CSS/color.css (hunk 4 above)
Comment (krishanu2911) [reply CSS/color.css]: Thank you for pointing out changed the font-weight which helps to differentiate

**Pull request merged**